tests/state/test_state_api.py

```diff
@@ -211,3 +211,12 @@ def test_get_states_page_size(self, size, stub_broker, api_client, state_middlew
             assert res.json["count"] == 10
         else:
             assert len(res.json["data"]) == size
+
+    def test_get_group_id(self, stub_broker, api_client, state_middleware):
```

Comment:
add 2 mesages to a group to be more exhaustive

add a test to test that before_process_message set group_id in the state